internal/action/find.go

```diff
@@ -141,13 +144,21 @@ func (s *Action) findSelection(ctx context.Context, c *cli.Context, choices []st
 	}
 }
 
-func filter(l []string, needle string) []string {
+func filter(l []string, needle string, regex bool) ([]string, error) {
 	choices := make([]string, 0, 10)
 	for _, value := range l {
-		if strings.Contains(strings.ToLower(value), needle) {
+		if regex {
+			matched, err := regexp.MatchString(needle, value)
```

Comment:
`regexp.MatchString` will compile the regular expression in every cycle of the loop. That could be pretty expensive. Please compile the RE outside of the loop and then use the compiled RE inside the loop. This will also simplify error handling.

Reply:
Compiling outside of the loop in: ecc5704cb582488e460718e9f7638fbcfaab1a82

Also replaced the `if else` with an early return.